enhancements/network/infiniband.md

```diff
@@ -0,0 +1,139 @@
+---
+title: infiniband
+authors:
+  - "@zshi"
+reviewers:
+  - "@pliurh"
+  - "@dougbtv"
+  - "@s1061123"
+  - "@danwinship"
+  - "@squeed"
+  - "@dcbw"
+  - "@yrobla"
+approvers:
+  - "@knobunc"
+  - "@fepan"
+creation-date: 2020-05-06
+last-updated: 2020-05-06
+status: implementable
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# InfiniBand
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [X] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+RDMA (Remote Direct Memory Access) fabrics such as InfiniBand and RDMA over
+Converged Ethernet (RoCE) are key to unlocking scalable performance for
+demanding Artificial Intelligence (AI) and Machine Learning (ML) applications.
+RoCE v2 is supported in SR-IOV Operator as Tech Preview since 4.3, This
+enhancement is to introduce native InfiniBand support.
+
+## Motivation
+
+Enable high-performance message transmission via RDMA for distributed workloads
+such as AI/ML.
+
+### Goals
+
+- Provide native InfiniBand support on pod additional network interfaces
+
+### Non-Goals
+
+- Support RoCE v2
+- Support GPUDirect RDMA
+- Support Ethernet over InfiniBand
+- Support IP over InfiniBand (IPoIB)
+- Support InfiniBand on pod primary network interface
+
+## Proposal
+
+Extend SR-IOV Operator to support both Ethernet and InfiniBand devices.
+
+SR-IOV Hardware NIC is managed by SR-IOV Operator and will be provisioned
+with a new device type `infiniBand` or `ib`.
+
+A new Container Network Interface (CNI) plugin `ib-sriov-cni` will be added.
+
+InfiniBand support will only be available on bare metal deployment.
+
+#### Hardware
+
+The following Mellanox NICs will be supported:
+- Mellanox Technologies MT27700 Family [ConnectX-4]
+- Mellanox Technologies MT28908 Family [ConnectX-6]
+
+#### InfiniBand SR-IOV CNI
+
+InfiniBand SR-IOV CNI `ib-sriov-cni` is a 3rd party CNI plugin that moves
+SR-IOV InfiniBand device to pod namespace and configures device attributes
```

Comment:
> that moves SR-IOV InfiniBand device to pod namespace

that moves SR-IOV InfiniBand IPoIB network device to the pod's network namespace

all ib-sriov-cni operations are either performed on the ipoib netdevice of the VF or the PF (with VF idx)

Reply:
What about IB device, does it only support IPoIB ?

Reply:
what is IB device ? you mean `mlx5_x` ?
my comment here is on ib-sriov-cni. (which works with IPoIB)

Reply:
If the application, running inside the container, is written to use RDMA communication natively instead of using IPoIB. Will it also be possible to use ib-sriov-cni to move device in pod namespace?

Reply:
RDMA application may or may not use the IB device associated IPoIB net device to exchange QP information to establish connection via RDMA_CM module. 

alternatively it may utilize another network interface and implement its own logic to exchange this information.

To the best of my knowledge if the GUID of the VF was preconfigured (and RDMA namespace isolation is not used) you would not need ib-sriov-cni for such applications (device plugin will provide the mounts)

Reply:
If I only have one additional VF attached to Pod, what will be `another network interface` used for exchanging QP information? I assume it will not be the default interface?
`ib-sriov-cni` has the option to configure GUID, if it's not used, do we need another component for the same GUID config?

Reply:
>  I assume it will not be the default interface?

it can be, why not ? 
or any other mean to exchange information between two RDMA applications

Reply:
> > I assume it will not be the default interface?
> 
> it can be, why not ?
Yes, I was thinking if there might be other components developed for exchanging information via kube resources.

> or any other mean to exchange information between two RDMA applications